DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateProgressController.cs

```diff
@@ -178,6 +181,24 @@ DelegateProgressAccessRoute accessedVia
             return RedirectToAction("Index", "ViewDelegate", new { delegateId });
         }
 
+        [HttpGet]
+        [Route("UnlockProgress")]
+        public IActionResult UnlockProgress(
+            int progressId,
+            int customisationId,
+            int delegateId,
+            DelegateProgressAccessRoute accessedVia
+        )
+        {
+            progressService.UnlockProgress(progressId);
+            notificationService.SendUnlockRequest(progressId);

```

Comment:
I don't think this is the correct behaviour. The ticket says nothing about send an email when we're actually doing the unlock, the unlock request just needs to direct the admin it is sent to to the correct page, where they can click the button for this action.

Reply:
ah I see so an admin should be able to unlock but a non admin will have to request an unlock? But this ticket is just for an admin being able to unlock? and changing the link in the email sent to an admin when a non admin requests an unlock is that right?

Reply:
Not exactly, they're from different parts of the system. The delegate accesses Learning Portal, then requests a course to be unlocked. This sends an email to an admin with the request (NotificationService.SendUnlockRequest), which links to the new page created here (or the old site equivalent) where they can then unlock the progress.